### PR TITLE
refactor(ng-update): handle namespaced angular decorators in update-tool

### PR DESCRIPTION
**Pull request by devversion**
Previously Angular did not support namespaced Angular decorators. e.g.

```ts
@core.Component({...})
@core.Directive({...})
```

Since this is now supported in ngtsc, we want to handle
such cases where stylesheets, templates are referenced in
namespaced decorators.

This can be achieved by moving the more advanced import
resolution logic made for the Hammer gesture migration
to the `cdk/update-tool`. Exposing these for other CDK
and Material migrations can be helpful anyway.

This also unveiled that we did not run non-misc test cases
automatically. This will be fixed as part of this commit.

### CODE REVIEW

File: src/cdk/schematics/update-tool/utils/imports.ts [hunk 1]
```diff
@@ -8,39 +8,106 @@
...
-  const decl = symbol.declarations[0];
-
-  if (!ts.isImportSpecifier(decl)) {
+  const declaration = symbol.declarations[0];
+  if (!ts.isImportSpecifier(declaration)) {
     return null;
   }
-
-  // Since "decl" is an import specifier, we can walk up three times to get a reference
-  // to the import declaration node (NamedImports -> ImportClause -> ImportDeclaration).
```
Comment (jelbourn) [inline src/cdk/schematics/update-tool/utils/imports.ts]: Why delete this comment?

File: src/cdk/schematics/update-tool/utils/imports.ts [hunk 2]
```diff
@@ -8,39 +8,106 @@
...
+  /** Module name from which the symbol has been imported. */
+  moduleName: string;
+}
+
+
+/** Resolves the import of the specified identifier. */
+export function getImportOfIdentifier(node: ts.Identifier, typeChecker: ts.TypeChecker): Import|
     null {
-  const symbol = typeChecker.getSymbolAtLocation(node);
+  const directImport = getSpecificImportOfIdentifier(node, typeChecker);
```
Comment (jelbourn) [inline src/cdk/schematics/update-tool/utils/imports.ts]: Add some comments here with examples of the different cases (e.g. a "direct import" vs. a "qualified import")?

File: src/cdk/schematics/update-tool/utils/imports.ts (hunk 1 above)
Comment (devversion) [reply src/cdk/schematics/update-tool/utils/imports.ts]: Not intentional. I just moved the whole file from the v9 hammer migration and this apparently didn't have that comment. Re-added.

File: src/cdk/schematics/update-tool/utils/imports.ts (hunk 2 above)
Comment (devversion) [reply src/cdk/schematics/update-tool/utils/imports.ts]: Added.

**Pull request merged**